utility/src/main/java/org/oppia/android/util/logging/CurrentAppScreenNameWrapper.kt

```diff
@@ -0,0 +1,24 @@
+package org.oppia.android.util.logging
+
+import android.content.Intent
+import org.oppia.android.app.model.CurrentAppScreen
+import org.oppia.android.app.model.ScreenName
+import org.oppia.android.util.extensions.putProtoExtra
+import javax.inject.Inject
+
+private const val CURRENT_APP_SCREEN_NAME_KEY = "key"
+
+/** Utility to provide utilities related to wrapping the current app-screen name in an intent. */
+class CurrentAppScreenNameWrapper @Inject constructor() {
+
+  /** Returns an intent packed with [CurrentAppScreen]. */
+  fun getCurrentAppScreenNameIntent(screenName: ScreenName): Intent = Intent().apply {
+    this.putProtoExtra(
+      CURRENT_APP_SCREEN_NAME_KEY,
+      CurrentAppScreen.newBuilder().setScreenName(screenName).build()
+    )
+  }
+
+  /** Returns the intent-key that'll be used to retrieve [ScreenName]. */
+  fun getCurrentAppScreenNameIntentKey(): String = CURRENT_APP_SCREEN_NAME_KEY
```

Comment:
You could add a getter to retrieve the screen name for a given intent and keep the key private. That might actually be a bit cleaner.

Reply:
But even if we go that way, we'd still need the key to be public for `putProtoExtra` right?

Reply:
Not if we take in the intent object (since then the key is only needed by the decorator to add/retrieve it).

Reply:
good point, done.